src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx

```diff
@@ -0,0 +1,110 @@
+import { TextField, Typography } from "@material-ui/core";
+import VerticalSpacer from "@saleor/apps/components/VerticalSpacer";
+import RadioGroupField from "@saleor/components/RadioGroupField";
+import { GiftCardError } from "@saleor/fragments/types/GiftCardError";
+import { getGiftCardErrorMessage } from "@saleor/giftCards/GiftCardUpdatePage/messages";
+import { FormChange } from "@saleor/hooks/useForm";
+import {
+  GiftCardExpiryTypeEnum,
+  TimePeriodTypeEnum
+} from "@saleor/types/globalTypes";
+import React from "react";
+import { MessageDescriptor, useIntl } from "react-intl";
+
+import TimePeriodField from "../TimePeriodField/TimePeriodField";
+import { giftCardExpirySelectMessages as messages } from "./messages";
+import { useGiftCardExpirySelectStyles as useStyles } from "./styles";
+
+interface UntranslatedOption {
+  label: MessageDescriptor;
+  value: GiftCardExpiryTypeEnum;
+}
+
+const options: UntranslatedOption[] = [
+  {
+    label: messages.neverExpireLabel,
+    value: GiftCardExpiryTypeEnum.NEVER_EXPIRE
+  },
+  {
+    label: messages.expiryPeriodLabel,
+    value: GiftCardExpiryTypeEnum.EXPIRY_PERIOD
+  },
+  {
+    label: messages.expirationDateLabel,
+    value: GiftCardExpiryTypeEnum.EXPIRY_DATE
+  }
+];
+
+interface GiftCardExpirySelectProps {
+  change: FormChange;
+  expiryPeriodType: TimePeriodTypeEnum;
+  expiryPeriodAmount: string;
```

Comment:
Should it be a string? Perhaps we should cast it later to a string but it makes more sense to have it as a number to me?

Reply:
Is it also a String in schema? If not we should treat it as number everywhere and only cast it to String in the input itself

Reply:
Not, then adding this as a number and parsing to the text with select field 👍